skfem/assembly/basis/subdomain_facet_basis.py

```diff
@@ -0,0 +1,80 @@
+from typing import Optional, Tuple
+
+import numpy as np
+from numpy import ndarray
+from skfem.element import Element
+from skfem.mapping import Mapping
+from skfem.mesh import Mesh
+
+from .boundary_facet_basis import BoundaryFacetBasis
+from ..dofs import Dofs
+
+
+class SubdomainFacetBasis(BoundaryFacetBasis):
+
+    def __init__(self,
+                 mesh: Mesh,
+                 elem: Element,
+                 elements: ndarray,
+                 mapping: Optional[Mapping] = None,
+                 intorder: Optional[int] = None,
+                 quadrature: Optional[Tuple[ndarray, ndarray]] = None,
+                 side: int = 0,
+                 dofs: Optional[Dofs] = None):
+        # Used by .with_element()
+        self._elements = elements
+        self._side = side
+
+        elements = mesh.normalize_elements(elements)
+        # Get indices of all facets in the mesh
+        all_facets, all_counts = np.unique(mesh.t2f,
+                                           return_counts=True)
+        # Find the exterior ones
+        exterior_facets = all_facets[all_counts == 1]
+        # Get indices of all facets in the subdomain
+        subdomain_facets, sub_counts = np.unique(mesh.t2f[:, elements],
+                                                 return_counts=True)
+        # Don't want facets on the exterior boundary, even if they are on the
+        # subdomain boundary
```

Comment:
What is the rationale behind silently dropping exterior facets vs. raising an error? Can you see an use case where this behavior is useful?

Reply:
The rational was that you already have boundary facet basis, so it is possible to use 2 bases in a list to `asm` and include the exterior basis. The inverse of this is not as easy... if subdomain basis automatically _includes_ the exterior boundary, I didn't see an easy way to remove it.

In my E&M work, I do not expect to ever want the external boundary mixed with an interior boundary.